hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/DiscoveryJoiner.java

```diff
@@ -22,17 +22,31 @@
 import com.hazelcast.nio.Address;
 import com.hazelcast.spi.discovery.DiscoveryNode;
 import com.hazelcast.spi.discovery.integration.DiscoveryService;
+import com.hazelcast.util.concurrent.BackoffIdleStrategy;
+import com.hazelcast.util.concurrent.IdleStrategy;
 
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
 
 import static com.hazelcast.util.Preconditions.checkNotNull;
+import static java.lang.Integer.getInteger;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 public class DiscoveryJoiner
         extends TcpIpJoiner {
+    private static final int DEFAULT_MAXIMUM_WAITING_FOR_INITIAL_MEMBERS_SECONDS = 5;
+    private static final String WAITING_FOR_INITIAL_MEMBERS_SECONDS_PROP = "hazelcast.discovery.initial.join.seconds";
```

Comment:
what about using `hazelcast.max.wait.seconds.before.join` property for this purpose, instead of introducing a new one?

Reply:
`hazelcast.wait.seconds.before.join` maybe ?

Reply:
ah yes, that's more suitable.